pkg/provisioner/ironic/ironic.go

```diff
@@ -1734,46 +1734,41 @@ func (p *ironicProvisioner) PowerOn() (result provisioner.Result, err error) {
 			p.log.Info("waiting for power status to change")
 			return operationContinuing(powerRequeueDelay)
 		}
-		result, err = p.changePower(ironicNode, nodes.PowerOn)
-		switch err.(type) {
-		case nil:
-		case HostLockedError:
-		default:
-			return transientError(errors.Wrap(err, "failed to power on host"))
+		if ironicNode.LastError != "" && !force {
```

Comment:
@sadasu (maybe I missed some bits but) can you please clarify the reason to check `ironicNode.LastError` before invoking `changePower`? Couldn't be something to be managed just after checking `changePower`?

Reply:
@andfasano When the 1st call to changePower() resulted in operationContinuing() and when PowerOn() is called again and this time the Last Error is not empty. We do not want to continue to changePower() without reporting the contents of LastError.

Reply:
My point was that, if LastError is set just after a call to changePower(), maybe an immediate `p.getNode()` after it would be sufficient to inspect the node _without_ waiting another loop (with the possibility eventually to return immediately an operationFailed).

In any case, it seems that in the current code the `operationFailed` is invoked just once (the second time). If the problem persists, the subsequent invocations will skip it given that the force flag will result to be set (PowerManagementError)

Reply:
I don't know that there's any reason to think a LastError will show up immediately after calling changePower(). In fact, quite the opposite - I'd expect it to almost always fail because of a timeout.

When the force flag is passed we return success, which will result in `clearError()` being called so that next time around the force flag will be unset and we _will_ report an error if another one occurs. This is consistent with all of the other methods that take a force flag.

If you call changePower first then either you will never see an error that appears asynchronously (presumably all of them, since the purpose of LastError is recording asynchronous errors) or you would have to only call it when the force flag is set, which means you couldn't change the power without first flagging a power management error.

Reply:
Ok